test/Client/AzureManaged.Tests/DurableTaskSchedulerClientExtensionsTests.cs

```diff
@@ -0,0 +1,174 @@
+        Mock<IDurableTaskClientBuilder> mockBuilder = new Mock<IDurableTaskClientBuilder>();
+        mockBuilder.Setup(b => b.Services).Returns(services);
+        DefaultAzureCredential credential = new DefaultAzureCredential();
+
+        // Act
+        mockBuilder.Object.UseDurableTaskScheduler(ValidEndpoint, ValidTaskHub, credential);
+
+        // Assert
+        ServiceProvider provider = services.BuildServiceProvider();
+        IOptions<GrpcDurableTaskClientOptions>? options = provider.GetService<IOptions<GrpcDurableTaskClientOptions>>();
+        options.Should().NotBeNull();
+
+        // Validate the configured options
+        var clientOptions = provider.GetRequiredService<IOptions<DurableTaskSchedulerClientOptions>>().Value;
+        clientOptions.EndpointAddress.Should().Be(ValidEndpoint);
+        clientOptions.TaskHubName.Should().Be(ValidTaskHub);
+        clientOptions.Credential.Should().BeOfType<DefaultAzureCredential>();
+    }
+
+    [Fact]
+    public void UseDurableTaskScheduler_WithConnectionString_ShouldConfigureCorrectly()
+    {
+        // Arrange
+        ServiceCollection services = new ServiceCollection();
+        Mock<IDurableTaskClientBuilder> mockBuilder = new Mock<IDurableTaskClientBuilder>();
+        mockBuilder.Setup(b => b.Services).Returns(services);
+        string connectionString = $"Endpoint={ValidEndpoint};Authentication=DefaultAzure;TaskHub={ValidTaskHub}";
+
+        // Act
+        mockBuilder.Object.UseDurableTaskScheduler(connectionString);
+
+        // Assert
+        ServiceProvider provider = services.BuildServiceProvider();
+        IOptions<GrpcDurableTaskClientOptions>? options = provider.GetService<IOptions<GrpcDurableTaskClientOptions>>();
+        options.Should().NotBeNull();
+
+        // Validate the configured options
+        var clientOptions = provider.GetRequiredService<IOptions<DurableTaskSchedulerClientOptions>>().Value;
+        clientOptions.EndpointAddress.Should().Be(ValidEndpoint);
+        clientOptions.TaskHubName.Should().Be(ValidTaskHub);
+        clientOptions.Credential.Should().BeOfType<DefaultAzureCredential>();
+    }
+
+    [Theory]
+    [InlineData(null, "testhub")]
+    [InlineData("myaccount.westus3.durabletask.io", null)]
+    public void UseDurableTaskScheduler_WithNullParameters_ShouldThrowArgumentNullException(string endpoint, string taskHub)
+    {
+        // Arrange
+        ServiceCollection services = new ServiceCollection();
+        Mock<IDurableTaskClientBuilder> mockBuilder = new Mock<IDurableTaskClientBuilder>();
+        mockBuilder.Setup(b => b.Services).Returns(services);
+        DefaultAzureCredential credential = new DefaultAzureCredential();
+
+        // Act
+        Action action = () => mockBuilder.Object.UseDurableTaskScheduler(endpoint, taskHub, credential);
+
+        // Assert
+        action.Should().NotThrow(); // The validation happens when building the service provider
```

Comment:
The comment and action here do not align: the service provider is not built here and no options validation will have ran yet.

Reply:
this is unneccesary will remove